app/assets/javascripts/components/AccountMenu/ConfirmPassword.tsx

```diff
@@ -0,0 +1,225 @@
+          <span className="color-dark-red">
+            Standard Notes does not have a password reset option
+          </span>
+          . If you forget your password, you will permanently lose access to
+          your data.
+        </div>
+        <form onSubmit={handleConfirmFormSubmit} className="px-3 mb-1">
+          <InputWithIcon
+            className="mb-2"
+            icon="password"
+            inputType={showPassword ? 'text' : 'password'}
+            placeholder="Confirm password"
+            value={confirmPassword}
+            onChange={handlePasswordChange}
+            onKeyDown={handleKeyDown}
+            toggle={{
+              toggleOnIcon: 'eye',
+              toggleOffIcon: 'eye',
+              title: 'Show password',
+              onClick: (toggled) => setShowPassword(toggled),
+            }}
+            ref={passwordInputRef}
+            disabled={isRegistering}
+          />
+          <Button
+            className="btn-w-full mt-1 mb-3"
+            label={
+              isRegistering ? 'Creating account...' : 'Create account & sign in'
+            }
+            type="primary"
+            onClick={handleConfirmFormSubmit}
+            disabled={isRegistering}
+          />
+          <Checkbox
+            name="is-ephemeral"
+            label="Stay signed in"
+            checked={!isEphemeral}
+            onChange={handleEphemeralChange}
+            disabled={isRegistering}
+          />
+          {notesAndTagsCount > 0 ? (
+            <Checkbox
+              name="should-merge-local"
+              label={`Merge local data (${notesAndTagsCount} notes and tags)`}
+              checked={shouldMergeLocal}
+              onChange={handleShouldMergeChange}
+              disabled={isRegistering}
+            />
+          ) : null}
+        </form>
+        <div className="h-1px my-2 bg-border"></div>
+        <button
+          className="sn-dropdown-item font-bold"
+          onClick={() => {
+            setShowAdvanced(!showAdvanced);
+          }}
+        >
+          <div className="flex item-center">
+            Advanced options
+            <Icon
+              type="chevron-down"
+              className="sn-icon--small color-grey-1 ml-1"
+            />
+          </div>
+        </button>
+        {showAdvanced ? (
+          <div className="px-3 my-2">
+            <Checkbox
+              name="custom-sync-server"
+              label="Custom sync server"
+              checked={enableCustomServer}
+              onChange={handleEnableServerChange}
+              disabled={isRegistering}
+            />
+            <InputWithIcon
+              inputType="text"
+              icon="server"
+              placeholder="https://api.standardnotes.com"
+              value={syncServer}
+              onChange={handleSyncServerChange}
+              disabled={!enableCustomServer && !isRegistering}
+            />
+          </div>
+        ) : null}
```

Comment:
This Advanced options section is repeated both in the Create Account, Confirm Password and Sign In panes. Would be nice to move this to a common parent component to all of these steps, to avoid repeating code and having to pass the enableCustomServer prop up to the HOC and back down to each of the panes